server/server.go

```diff
@@ -510,17 +510,26 @@ func (s *Server) startMonitoring(secure bool) error {
 		hp           string
 		err          error
 		httpListener net.Listener
+		port   int
 	)
 
 	if secure {
-		hp = net.JoinHostPort(opts.HTTPHost, strconv.Itoa(opts.HTTPSPort))
+		port = opts.HTTPSPort
+		if port == -1 {
+			port = 0
+		}
```

Comment:
We have 2 different semantics... for the server.Port, 0 means default. for http.Port and https.Port, 0 means not set. I kept the -1 for random we use in server.Port and extended it to http(s).Port so that I didn't risk any existing config logic.